state.go

```diff
@@ -422,17 +452,26 @@ type AuthorizedState struct {
 	BaseState
 }
 
-func (a *AuthorizedState) Handle(c Controller) (State, bool) {
-	// TODO HasUpgrade should return update information
-	has, err := c.HasUpgrade()
+func (a *AuthorizedState) Handle(ctx *StateContext, c Controller) (State, bool) {
+	sd, err := LoadStateData(ctx.store)
 	if err != nil {
```

Comment:
No, I don't think so. If there's no update information (we get ErrNotExist) then we should treat it as if no update was in progress. 

I still need to complete the logic here. Looks like we'll need to involve image ID to figure out if we're really running the new image or not. I think that the logic may look like this, but I'd like to know your opinions @pasinskim @kacf 

```
LoadStateData()
if err != ErrNotExist 
    can't do anythng about it && raise error?
if last state in (MenderStateReboot, MenderStateInstall)
    if HasUpgrade()
        if currentImageID == UpdatedImageID() 
            update complete && report success
        else
            we either crashed and were restarted or update image
            is the same as previous??
    else
        so update was being installed (or was installed) but upgrade_avaialble was not flipped, 
        hence we're running the old image, start from scratch
else
    ignore previous update, start from scratch

RemoveStateData()
```